spec/models/user_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  user = User.new(name: "Me", email: "danijuyusuf@mail.com")
+  context "User" do
+    it "returns true if user is valid" do
+      expect(user.valid?).to be true
+    end
+
+    it "has email which cannot be empty" do
+      user.email = ""
+      expect(user.valid?).to be false
+    end
+
+    it "has name which cannot be empty" do
+      user.name = ""
+      expect(user.valid?).to be false
+    end
+
+    it "has name of length less than 50" do
+      expect(user.name.length).to be <= 50
+    end
+
+    it "has email of length less than 30" do
+      expect(user.email.length).to be <= 30
+    end
+
+    it "has unique email" do
+      duplicate_user = user.dup
+      duplicate_user_email = user.email
+      expect(duplicate_user.valid?).to be false
+    end
+
+    it "has unique email" do
+      duplicate_user = user.dup
+      duplicate_user_email = user.email.upcase
```

Comment:
Useless assignment to variable - `duplicate_user_email`. Did you mean `duplicate_user`?